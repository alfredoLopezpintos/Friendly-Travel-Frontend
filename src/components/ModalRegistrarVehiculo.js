import React, { useState, useEffect } from "react";
import { URLS } from "../utils/urls";
import axios from "axios";
import {
    Dialog,
    DialogTitle,
    DialogContent,
    DialogActions,
    TextField,
    Checkbox,
    FormControlLabel,
    Button,
    Grid,
    Snackbar,
    FormHelperText,
    Typography
} from "@material-ui/core";
import { checkPlate, checkVehicleYear, checkSeats } from "../utils/ValidationFunctions";
import { getToken } from "./service/AuthService";
import { useImageUploader } from "./service/ImageUploader";
// import { Button as Button2 } from "../components/Button";
import { Button as Button2 }  from '@rodrisu/friendly-ui/build/button';
import "./pages/ChangeData.css"

export default function ModalRegistrarVehiculo() {
    const [manufacturer, setManufacturer] = useState('');
    const [model, setModel] = useState('');
    const [year, setYear] = useState('');
    const [airBag, setAirBag] = useState(false);
    const [airCond, setAirCond] = useState(false);
    const [seats, setSeats] = useState('');
    const [plate, setPlate] = useState('');
    const [file, setFile] = useState(null);
    const [isSubmitting, setIsSubmitting] = useState(false);
    const [result, setResult] = useState(null);
    const [isFormSubmitted, setIsFormSubmitted] = useState(false);
    const { image, onFileChange, removeImage, uploadImage } = useImageUploader();
    const [imageError, setImageError] = useState(false);
    const [snackbarOpen, setSnackbarOpen] = useState(false);

    const [displayModal, setDisplayModal] = useState(false);
    const handleClickOpen = () => {
        setDisplayModal(true);
      };
    
      const handleClose = () => {
        setDisplayModal(false);
      };

    const requestConfig = {
        headers: {
            Authorization: JSON.parse(getToken()),
            "Content-Type": "application/json",
        },
    };

    const handleSubmit = async (e) => {
        e.preventDefault();
        setIsFormSubmitted(true);

        // Validation logic
        let validYear = false;
        validYear = checkVehicleYear(year).valid;

        let validPlate = false;
        validPlate = checkPlate(plate).valid;

        let validSeats = false;
        validSeats = checkSeats(seats).valid;

        if (!image) {
            setImageError(true);
            setIsSubmitting(false);
            return;
        }

        if (manufacturer && model && validYear && seats && validPlate && validSeats && !imageError) {
            setIsSubmitting(true);
            const requestBody = {
                manufacturer,
                model,
                year,
                airbag: airBag,
                airCond: airCond,
                plate,
                seats,
            };

            if (image) {
                requestBody.image = image;
            }

            try {
                const imageKey = await uploadImage(plate);
                requestBody.imageKey = imageKey;

                //console.log(URLS.POST_VEHICLE_URL)
                //console.log(JSON.stringify(requestBody))
                //console.log(requestConfig)

                const responseAddVehicle = await axios.post(
                    URLS.POST_VEHICLE_URL,
                    JSON.stringify(requestBody),
                    requestConfig
                );
                setResult(responseAddVehicle.data.message);
                setSnackbarOpen(true);
                setIsSubmitting(false);
            } catch (error) {
                console.error(error);
                setResult(error.response.data.message);
                setIsSubmitting(false);
                setSnackbarOpen(true);
            }
        } else {
            setIsSubmitting(false);
            setResult(
                'Por favor complete todos los campos y asegúrese que todos tienen el formato correcto.'
            );
        }
    };

    const isManufacturerEmpty = isFormSubmitted && manufacturer === '';
    const isModelEmpty = isFormSubmitted && model === '';
    const isYearEmpty = isFormSubmitted && year === '';
    const isSeatsEmpty = isFormSubmitted && seats === '';
    const isPlateEmpty = isFormSubmitted && plate === '';
    const isImageEmpty = isFormSubmitted && file === null;

    // Error message for invalid year
    let yearHelperText = '';
    if (isYearEmpty) {
        yearHelperText = 'Campo requerido';
    } else if (year !== '' && !checkVehicleYear(year).valid) {
        yearHelperText = checkVehicleYear(year).message;
    }

    // Error message for invalid license plate
    let plateHelperText = '';
    if (isPlateEmpty) {
        plateHelperText = 'Campo requerido';
    } else if (plate !== '' && !checkPlate(plate).valid) {
        plateHelperText = checkPlate(plate).message;
    }

    // Error message for invalid seats
    let seatsHelperText = '';
    if (isSeatsEmpty) {
        seatsHelperText = 'Campo requerido';
    } else if (seats !== '' && !checkSeats(seats).valid) {
        seatsHelperText = checkSeats(seats).message;
    }

    const handleSnackbarClose = () => {
        setSnackbarOpen(false);
        if (result === 'Vehículo agregado correctamente.') {
            handleClose();
        }
    };

    return (
        <>
            <Button2 
              className="btns"
              buttonStyle="btn--test"
              buttonSize="btn--large"
              onClick={handleClickOpen}>
              Agregar Vehículo
            </Button2>
            <Dialog open={displayModal} onClose={handleClose} data-testid="form">
                <form onSubmit={handleSubmit}>
                    <DialogTitle>Agregar vehículo</DialogTitle>
                    <DialogContent>
                        <Grid container direction="column" alignItems="stretch" spacing={1}>
                            <Grid item>
                                <TextField
                                    data-testid="camporequerido"
                                    label="Marca"
                                    id="Marca"
                                    value={manufacturer}
                                    onChange={(e) => setManufacturer(e.target.value)}
                                    fullWidth
                                    error={isManufacturerEmpty}
                                    helperText={isManufacturerEmpty && <FormHelperText>Campo requerido</FormHelperText>}
                                />
                            </Grid>
                            <Grid item>
                                <TextField
                                    label="Modelo"
                                    id="Modelo"
                                    value={model}
                                    onChange={(e) => setModel(e.target.value)}
                                    fullWidth
                                    error={isModelEmpty}
                                    helperText={isModelEmpty && <FormHelperText>Campo requerido</FormHelperText>}
                                />
                            </Grid>
                            <Grid item>
                                <TextField
                                    label="Año"
                                    id="Año"
                                    value={year}
                                    onChange={(e) => setYear(e.target.value)}
                                    fullWidth
                                    error={(isFormSubmitted && isYearEmpty) || (isFormSubmitted && yearHelperText !== '')}
                                    helperText={yearHelperText}
                                />
                            </Grid>
                            <Grid item>
                                <Grid container direction="row">
                                    <Grid item style={{ marginRight: '20px' }}>
                                        <FormControlLabel
                                            control={<Checkbox value={airBag} onChange={(e) => setAirBag(e.target.checked)} />}
                                            label="Airbag"
                                            id="Airbag"
                                        />
                                    </Grid>
                                    <Grid item>
                                        <FormControlLabel
                                            control={<Checkbox value={airCond} onChange={(e) => setAirCond(e.target.checked)} />}
                                            label="A/C"
                                            id="A/C"
                                        />
                                    </Grid>
                                </Grid>
                            </Grid>
                            <Grid item>
                                <TextField
                                    label="Asientos"
                                    id="Asientos"
                                    value={seats}
                                    onChange={(e) => setSeats(e.target.value)}
                                    fullWidth
                                    error={(isFormSubmitted && isSeatsEmpty) || (isFormSubmitted && seatsHelperText !== '')}
                                    helperText={seatsHelperText}
                                />
                            </Grid>
                            <Grid item>
                                <TextField
                                    label="Matrícula"
                                    id="Matrícula"
                                    value={plate}
                                    onChange={(e) => setPlate(e.target.value)}
                                    fullWidth
                                    error={(isFormSubmitted && isPlateEmpty) || (isFormSubmitted && plateHelperText !== '')}
                                    helperText={plateHelperText}
                                />
                            </Grid>
                            <Grid item>
                                <label htmlFor="upload-file" style={{ display: 'block', marginTop: '15px' }}>
                                    <Typography variant="subtitle2">Propiedad del vehículo</Typography>
                                </label>
                                {!image && (
                                    <TextField
                                        data-testid="upload-file"
                                        id="upload-file"
                                        type="file"
                                        accept="image/*"
                                        onChange={onFileChange}
                                        error={imageError}
                                        helperText={imageError && <FormHelperText error={true}>Campo requerido</FormHelperText>}
                                        fullWidth
                                    />
                                )}
                                {image && (
                                    <div>
                                        <div><img src={image} alt="Uploaded file" className="uploaded-image" /></div>
                                        <Button onClick={removeImage}>Quitar archivo</Button>
                                    </div>
                                )}
                            </Grid>
                        </Grid>
                    </DialogContent>
                    <DialogActions>
                        <Button onClick={handleClose} color="primary">
                            Cancelar
                        </Button>
                        <Button type="submit" color="primary" disabled={isSubmitting}>
                            {isSubmitting ? 'Agregando...' : 'Agregar'}
                        </Button>
                    </DialogActions>
                </form>
                <Snackbar
                anchorOrigin={{ vertical: "bottom", horizontal: "center" }}
                open={snackbarOpen}
                autoHideDuration={5000}
                onClose={handleSnackbarClose}
                message={result}
                contentProps={{
                    style: { backgroundColor: 'green' },
                }}
            />
            </Dialog>
        </>
    );
}
