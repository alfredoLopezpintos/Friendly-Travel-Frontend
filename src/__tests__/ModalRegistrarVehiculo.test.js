import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import axios from "axios";
import ModalRegistrarVehiculo from '../components/ModalRegistrarVehiculo';

jest.mock("axios");

describe('ModalRegistrarVehiculo component', () => {
    let toggleModalMock;

    beforeEach(() => {
        toggleModalMock = jest.fn();
        render(
            <ModalRegistrarVehiculo displayModal={true} toggleModal={toggleModalMock} />
        );
    });

    afterEach(() => {
        toggleModalMock.mockReset();
    });

    // test("should add a vehicle when submitting a valid form", async () => {
    //     // setting up mock response from server
    //     const vehicle = { id: "1", marca: "Toyota", modelo: "Corolla", anho: "2015" };
    //     axios.post.mockImplementationOnce(() => Promise.resolve({ data: vehicle }));

    //     const marcaInput = screen.getByLabelText("Marca");
    //     fireEvent.change(marcaInput, { target: { value: "Toyota" } });

    //     const modeloInput = screen.getByLabelText("Modelo");
    //     fireEvent.change(modeloInput, { target: { value: "Corolla" } });

    //     const anhoInput = screen.getByLabelText("Año");
    //     fireEvent.change(anhoInput, { target: { value: "2015" } });

    //     const airbagCheckbox = screen.getByLabelText("Airbag");
    //     fireEvent.click(airbagCheckbox);

    //     const aircondCheckbox = screen.getByLabelText("A/C");
    //     fireEvent.click(aircondCheckbox);

    //     const asientosInput = screen.getByLabelText("Asientos");
    //     fireEvent.change(asientosInput, { target: { value: "5" } });

    //     const matriculaInput = screen.getByLabelText("Matrícula");
    //     fireEvent.change(matriculaInput, { target: { value: "ABC123" } });

    //     const uploadFileInput = screen.getByLabelText("Propiedad del vehículo");
    //     fireEvent.change(uploadFileInput, { target: { files: [new File([], "file.png")] } });

    //     const agregarBtn = screen.getByText("Agregar");
    //     fireEvent.click(agregarBtn);

    //     expect(agregarBtn).toBeInTheDocument("Agregando...");

    //     await waitFor(() => {
    //         expect(agregarBtn).toHaveTextContent("Agregar");
    //         expect(screen.queryByText("Vehículo agregado correctamente.")).toBeTruthy();
    //     });
    // });

    // test("should show an error snackbar when there was an error adding the vehicle", async () => {
    //     // setting up mock response from server
    //     axios.post.mockImplementationOnce(() => Promise.reject({ response: { data: { error: "failed request" } } }));

    //     const marcaInput = screen.getByLabelText("Marca");
    //     fireEvent.change(marcaInput, { target: { value: "Toyota" } });

    //     const modeloInput = screen.getByLabelText("Modelo");
    //     fireEvent.change(modeloInput, { target: { value: "Corolla" } });

    //     const anhoInput = screen.getByLabelText("Año");
    //     fireEvent.change(anhoInput, { target: { value: "2015" } });

    //     const airbagCheckbox = screen.getByLabelText("Airbag");
    //     fireEvent.click(airbagCheckbox);

    //     const aircondCheckbox = screen.getByLabelText("A/C");
    //     fireEvent.click(aircondCheckbox);

    //     const asientosInput = screen.getByLabelText("Asientos");
    //     fireEvent.change(asientosInput, { target: { value: "5" } });

    //     const matriculaInput = screen.getByLabelText("Matrícula");
    //     fireEvent.change(matriculaInput, { target: { value: "ABC123" } });

    //     const uploadFileInput = screen.getByLabelText("Propiedad del vehículo");
    //     fireEvent.change(uploadFileInput, { target: { files: [new File([], "file.png")] } });

    //     const agregarBtn = screen.getByText("Agregar");
    //     fireEvent.click(agregarBtn);

    //     expect(agregarBtn).toBeInTheDocument("Agregando...");

    //     await waitFor(() => {
    //         expect(agregarBtn).toHaveTextContent("Agregar");
    //         expect(screen.getByTestId("snackbar")).toBeInTheDocument();
    //     });
    // });

    test('should render successfully', async () => {
        const formElement = screen.getByTestId('form');
        await waitFor(() => expect(formElement).toMatchSnapshot());
    });

    test('should close the modal when cancel button is clicked', async () => {
        const cancelButton = screen.getByText('Cancelar');
        fireEvent.click(cancelButton);
        await waitFor(() =>
            expect(toggleModalMock).toHaveBeenCalledTimes(1)
        );
    });

    test('submitting form data with missing fields should display error messages', async () => {

        const submitButton = screen.getByText('Agregar');
        fireEvent.click(submitButton);

        await waitFor(() => {
            expect(screen.getByTestId('camporequerido')).toBeInTheDocument();
        });

        expect(screen.getByTestId('form')).toBeInTheDocument();
    });

    test('submitting form data with invalid plate should display error message', async () => {

        fireEvent.change(screen.getByLabelText('Matrícula'), { target: { value: 'abc-123' } });

        const submitButton = screen.getByText('Agregar');
        fireEvent.click(submitButton);

        await waitFor(() => {
            expect(screen.queryByText('Ingrese una matrícula válida')).toBeInTheDocument();
        });

        expect(screen.getByTestId('form')).toBeInTheDocument();
    });

    test('should show error message when year is not valid', async () => {
        const inputYear = screen.getByLabelText('Año');
        fireEvent.change(inputYear, { target: { value: '123456' } });
        const addButton = screen.getByText('Agregar');
        fireEvent.submit(addButton);

        await waitFor(() =>
            expect(screen.getByText('Por favor ingrese un año entre 1950 y 2024').toBeInTheDocument)
        );
    });

    test('should show error message when seats is not valid', async () => {
        const inputSeats = screen.getByLabelText('Asientos');
        fireEvent.change(inputSeats, { target: { value: '0' } });
        const addButton = screen.getByText('Agregar');
        fireEvent.submit(addButton);

        await waitFor(() => {
            expect(screen.getByText('El número de asientos debe ser entre 1 y 4')).toBeInTheDocument();
        });
    });

    // test('submitting valid form data should add a new vehicle', async () => {
    //     const mockUploadImage = jest.fn(() => Promise.resolve('imageKey'));
    //     const mockGetToken = jest.fn(() => JSON.stringify('token'));
    //     const mockPost = jest.fn().mockResolvedValue({ data: { message: 'Vehicle added successfully' } });


    //     jest.mock('../components/service/ImageUploader', () => ({
    //       useImageUploader: () => ({
    //         image: 'image',
    //         onFileChange: jest.fn(),
    //         removeImage: jest.fn(),
    //         uploadImage: mockUploadImage,
    //       }),
    //     }));

    //     jest.mock('../components/service/AuthService', () => ({
    //       getToken: mockGetToken,
    //     }));

    //     jest.mock('axios', () => ({
    //         create: () => ({
    //           post: mockPost,
    //         }),
    //       }));

    //     fireEvent.change(screen.getByLabelText('Marca'), { target: { value: 'Toyota' } });
    //     fireEvent.change(screen.getByLabelText('Modelo'), { target: { value: 'Corolla' } });
    //     fireEvent.change(screen.getByLabelText('Año'), { target: { value: '2021' } });
    //     fireEvent.click(screen.getByLabelText('Airbag'));
    //     fireEvent.click(screen.getByLabelText('A/C'));
    //     fireEvent.change(screen.getByLabelText('Asientos'), { target: { value: '4' } });
    //     fireEvent.change(screen.getByLabelText('Matrícula'), { target: { value: 'ABC1234' } });

    //     const submitButton = screen.getByText('Agregar');
    //     fireEvent.click(submitButton);

    //     await waitFor(() => expect(mockPost).toHaveBeenCalledTimes(1));
    //     expect(axios.post).toHaveBeenCalledWith(
    //       'https://example.com/api/vehicles',
    //       JSON.stringify({
    //         manufacturer: 'Toyota',
    //         model: 'Corolla',
    //         year: '2021',
    //         airBag: true,
    //         airCond: true,
    //         plate: 'ABC1234',
    //         seats: '4',
    //         image: 'image',
    //         imageKey: 'imageKey',
    //       }),
    //       {
    //         headers: {
    //           Authorization: 'token',
    //           'Content-Type': 'application/json',
    //         },
    //       }, { timeout: 5000 }
    //     );

    //     expect(screen.getByTestId('form')).not.toBeInTheDocument();
    //   });


});