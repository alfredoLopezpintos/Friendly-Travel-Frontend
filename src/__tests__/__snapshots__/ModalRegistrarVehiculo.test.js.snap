// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalRegistrarVehiculo component should render successfully 1`] = `
<div
  class="MuiDialog-root"
  data-testid="form"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper"
    role="none presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
      role="dialog"
    >
      <form>
        <div
          class="MuiDialogTitle-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          >
            Agregar vehículo
          </h2>
        </div>
        <div
          class="MuiDialogContent-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                data-testid="camporequerido"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="Marca"
                  id="Marca-label"
                >
                  Marca
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="Marca"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="Modelo"
                  id="Modelo-label"
                >
                  Modelo
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="Modelo"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="Año"
                  id="Año-label"
                >
                  Año
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="Año"
                    type="text"
                    value=""
                  />
                </div>
                
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                  style="margin-right: 20px;"
                >
                  <label
                    class="MuiFormControlLabel-root"
                    id="Airbag"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-4"
                          data-indeterminate="false"
                          type="checkbox"
                          value="false"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      Airbag
                    </span>
                  </label>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <label
                    class="MuiFormControlLabel-root"
                    id="A/C"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-4"
                          data-indeterminate="false"
                          type="checkbox"
                          value="false"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      A/C
                    </span>
                  </label>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="Asientos"
                  id="Asientos-label"
                >
                  Asientos
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="Asientos"
                    type="text"
                    value=""
                  />
                </div>
                
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="Matrícula"
                  id="Matrícula-label"
                >
                  Matrícula
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="Matrícula"
                    type="text"
                    value=""
                  />
                </div>
                
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <label
                for="upload-file"
                style="display: block; margin-top: 15px;"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2"
                >
                  Propiedad del vehículo
                </h6>
              </label>
              <div
                accept="image/*"
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                data-testid="upload-file"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="upload-file"
                    type="file"
                    value=""
                  />
                </div>
              </div>
              
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Cancelar
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Agregar
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
